db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -627,7 +627,7 @@ Status DBImpl::AtomicFlushMemTablesToOutputFiles(
       if (sfm) {
         std::string file_path = MakeTableFileName(
             cfds[i]->ioptions()->cf_paths[0].path, file_meta[i].fd.GetNumber());
-        sfm->OnAddFile(file_path);
+        s = sfm->OnAddFile(file_path);
```

Comment:
Stress test now sometimes shows a problem and I suspect that this line is related. Here is the failure:
```
2020/12/26-14:41:51.516292 14b0efeeb700 [ERROR] [/db_impl/db_impl_compaction_flush.cc:2521] Waiting after background flush error: IO error: No such file or directorywhile stat a file for size: /dev/shm/rocksdb/rocksdb_c
rashtest_whitebox/000000.sst: No such file or directoryAccumulated background error counts: 1
```
My theory is that, in the case where the memtable is empty, there is no file flushed but here the registration is called anyway. It used to be that sfm->OnAddFile() rejected it since the file doesn't exist, but now it fails the flush, which is unexpected. If we agree that it is the problem, we should fix it.

Reply:
@siying There is a comparable line above (line 287).  The errors could be ignored if required or is there a way to test that the file is not empty/was flushed?  How do we write a test to show this failure?

Reply:
If atomic flush is enabled, SFM is enabled, and there are more than one CF with all of their memtables to be empty, a Flush() call might trigger this issue.

A better fix might be that if no file is generated, we should not go into this code path. We probably can know whether a file is created (file_meta[i] is ever updated) in some way.